src/query.js

```diff
@@ -34,12 +34,18 @@ function query (data, params = {}) {
 
   // ArcGIS client warnings
   if (options.toEsri) {
-    if (!hasIdField) console.warn(`The requested provider has no "idField" assignment. This can cause errors in ArcGIS clients`)
+    if (!hasIdField) console.warn(`WARNING: requested provider has no "idField" assignment. This can cause errors in ArcGIS clients`)
     else if (data.metadata.idField.toLowerCase() === 'objectid' && data.metadata.idField !== 'OBJECTID') {
-      console.warn(`The requested provider's "idField" is a mixed-case version of "OBJECTID". This can cause errors in ArcGIS clients`)
-    } else if (queriedData.features.some(feature => { return !Number.isInteger(feature.attributes.OBJECTID) || feature.attributes.isArrayOBJECTID > 2147483647 })) {
-      console.warn(`OBJECTIDs created from provider's "idField" are not integers from 0 to 2147483647`)
+      console.warn(`WARNING: requested provider's "idField" is a mixed-case version of "OBJECTID". This can cause errors in ArcGIS clients`)
     }
+    if (queriedData.features && queriedData.features.some(feature => { return !Number.isInteger(feature.attributes.OBJECTID) || feature.attributes.OBJECTID > 2147483647 })) {
```

Comment:
this is a pretty expensive check, I think you'd be better off pushing this code into Winnow

Reply:
Great catch, moved to winnow https://github.com/koopjs/winnow/pull/76.